src/ui/Base/RegisteredNamedMethods.ts

```diff
@@ -447,6 +448,68 @@ Initialization.registerNamedMethod('configureResourceRoot', (path: string) => {
   configureResourceRoot(path);
 });
 
+function findAllComponentsOfType<T extends BaseComponent>(type: { new (...args: any[]): T }, searchRoot = document.body) {
```

Comment:
Could you use the method **getCoveoAnalytics** around line 224 to get the Analytics component?
I don't think you have to manage the case where you have to manage multiple Analytics component in the same page but I might be mistaken. @olamothe

Reply:
Good point. I was trying to make sure that if somebody mistakenly used more than one Analytics object, they would all be disabled. However, that is a bit too much of a waste of performance just for this reason.

Resolved by using `getCoveoAnalytics` instead.